Vagrantfile

```diff
@@ -1,71 +1,302 @@
+
+      echo 'export LIBRARY_PATH=/opt/crystal/lib/crystal/lib/:${LIBRARY_PATH}' >> /etc/profile.d/crystal.sh
+      echo 'export PATH=/opt/crystal/bin:$PATH' >> /etc/profile.d/crystal.sh
+    )
+  else
+    case family
+    when "ubuntu"
+      register_crystal_key =
+        if dist == "precise" || dist == "trusty"
+          %(
+            apt-key adv --keyserver keys.gnupg.net --recv-keys 09617FD37CC06B54
+          )
+        end
 
       c.vm.provision :shell, inline: %(
+        #{register_crystal_key}
+
         curl -s https://dist.crystal-lang.org/apt/setup.sh | sh
-        apt-get install -y crystal git libgmp3-dev zlib1g-dev libedit-dev libxml2-dev libssl-dev libyaml-dev libreadline-dev g++
-        curl -s https://crystal-lang.s3.amazonaws.com/llvm/llvm-3.5.0-1-linux-`uname -m`.tar.gz | tar xz -C /opt
-        echo 'export LIBRARY_PATH="/opt/crystal/embedded/lib"' > /etc/profile.d/crystal.sh
-        echo 'export PATH="$PATH:/opt/llvm-3.5.0-1/bin"' >> /etc/profile.d/crystal.sh
+        apt-get install -y crystal
+
+        echo 'export LIBRARY_PATH="/usr/lib/crystal/lib/"' >> /etc/profile.d/crystal.sh
       )
+    when "debian"
+      register_crystal_key =
+        if dist == "stretch"
+          %(
+            apt-key adv --keyserver keys.gnupg.net --recv-keys 09617FD37CC06B54
+          )
+        end
 
-      clone_crystal_from_vagrant.call(c)
-    end
-  end
+      c.vm.provision :shell, inline: %(
+        #{register_crystal_key}
+
+        curl -s https://dist.crystal-lang.org/apt/setup.sh | sh
+        apt-get install -y crystal
+
+        echo 'export LIBRARY_PATH="/usr/lib/crystal/lib/"' >> /etc/profile.d/crystal.sh
+      )
+    when "freebsd"
+      c.vm.post_up_message = <<-MSG
+
+      To install crystal from sources:
 
-  [[:jessie, '3.5'], [:stretch, '3.9']].product([32, 64]).each do |(dist, ver), bits|
-    box_name = "#{dist}#{bits}"
+      $ sudo portsnap fetch extract
+      $ sudo make -C/usr/ports/lang/crystal/ reinstall clean BATCH=yes
+      $ sudo make -C/usr/ports/devel/shards/ reinstall clean BATCH=yes
 
-    config.vm.define(box_name) do |c|
-      c.vm.box = "debian/#{box_name}"
+      To install pre-built crystal:
 
+      $ sudo pkg install -y crystal shards
+
+      Compile crystal with:
+
+      $ cd ~/crystal
+      $ gmake crystal
+
+      MSG
+    when "alpine"
       c.vm.provision :shell, inline: %(
-        apt-get -y install apt-transport-https dirmngr
-        apt-key adv --keyserver keys.gnupg.net --recv-keys 09617FD37CC06B54
-        echo 'deb https://dist.crystal-lang.org/apt crystal main' > /etc/apt/sources.list.d/crystal.list
-        apt-get update
-        apt-get -y install crystal curl git libgmp3-dev zlib1g-dev libedit-dev libxml2-dev libssl-dev libyaml-dev libreadline-dev g++ llvm-#{ver} llvm-#{ver}-dev
-        echo 'export LIBRARY_PATH="/opt/crystal/embedded/lib"' > /etc/profile.d/crystal.sh
+        apk add --no-cache crystal shards
```

Comment:
```suggestion
        apk update
        apk add --no-cache crystal shards
```

Reply:
or `apk add --update`

Reply:
I left a single `apk update` in the first provision